list/list.js

```diff
@@ -1004,8 +1005,8 @@ assign(DefineList.prototype, {
 	map: function(callback, thisArg) {
 		var mappedList = [],
 			self = this;
-		this.each(function(item, index, list) {
-			var mapped = callback.call(thisArg | self, item, index, self);
+		canReflect.eachIndex(this, function(item, index, list) {
+			var mapped = canReflect.call(callback, thisArg | self, item, index, self);
```

Comment:
The only thing I had in mind was things like a call expression. But it might not actually be useful because it represents a value, not a method that can be called.